vortex-java-sdk/vortex-java-sdk-consoleconnect/src/main/java/com/consoleconnect/vortex/cc/CCHttpClient.java

```
@@ -0,0 +1,135 @@
+package com.consoleconnect.vortex.cc;
+
+import com.consoleconnect.vortex.cc.model.CCClientProperty;
+import com.consoleconnect.vortex.cc.model.Member;
+import com.consoleconnect.vortex.cc.model.Role;
+import com.consoleconnect.vortex.cc.model.UserInfo;
+import com.consoleconnect.vortex.core.exception.VortexException;
+import com.consoleconnect.vortex.core.toolkit.GenericHttpClient;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.MapUtils;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+
+@Slf4j
+public class CCHttpClient {
+
+  private final CCClientProperty clientProperty;
+  private final GenericHttpClient genericHttpClient;
+
+  private final Cache<String, Member> memberLoadingCache =
+      CacheBuilder.newBuilder().maximumSize(1000).expireAfterWrite(2, TimeUnit.MINUTES).build();
+
+  public CCHttpClient(CCClientProperty clientProperty, GenericHttpClient genericHttpClient) {
+    this.clientProperty = clientProperty;
+    this.genericHttpClient = genericHttpClient;
+  }
+
+  public void assignRole2Member(String username, String role) {
+
+    String url =
+        String.format(
+            CCEndpoints.UPDATE_MEMBER_ROLES, clientProperty.getCompanyUsername(), username, role);
+
+    try {
+      genericHttpClient.blockPut(
+          clientProperty.getBaseUrl() + url,
+          Map.of(clientProperty.getApiKeyName(), clientProperty.getAdminApiKey()),
+          null,
+          new ParameterizedTypeReference<>() {});
+    } catch (WebClientResponseException.Conflict webClientResponseException) {
+      log.warn("Role has exist, username:{}", username);
```

Pls double check if you can ignore this exception or handle this exception as a normal case

---

If the role has been added, this exception will be thrown. I think we can add a warning log.